rust-guest/src/context/context.rs

```diff
@@ -0,0 +1,102 @@
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+use lazy_static::lazy_static;
+use std::sync::Mutex;
+
+use std::io::Cursor;
+use std::mem;
+use std::collections::HashMap;
+use super::generated::{Encode, Decode, Context, Request, Response};
+
+lazy_static! {
+    pub static ref PTR_LEN: Mutex<(u32, u32)> = Mutex::new((0, 0));
+}
+
+pub trait RunContext {
+    fn new() -> Self;
+    fn from_read_buffer(self, read_buff_global: &mut Cursor<&mut Vec<u8>>) -> Self;
```

Comment:
Does from_read_buffer need to have the global buffer passed in? Is there any way for the function argument to use a static variable with a polyglot buffer?

Reply:
yeah, I am experimenting with this via tossing the buffer in the global mutex, and getting rid of the PTR/LEN Vec reconstruction approach. Will update this function interface to not pass in in this way in if that works as intended.

Reply:
So after experimenting, I still think this is the cleanest approach. Using globals with our current implementation means unlocking every time even a wrapped immutable ref is borrowed. Other global implementations (like using a `Rc<RefCell<T>>` pattern) required much more boilerplate up front compared with lazy_static and much clunkier to use imo.